.github/workflows/build.yaml

```diff
@@ -35,9 +40,18 @@ jobs:
       # if unit tests fail to import llsd (i.e. wrong Python interpreter),
       # make py.exe enumerate the possibilities and explain its choice
       PYLAUNCHER_DEBUG: "1"
+      GIT_REF: ${{ github.head_ref || github.ref }}
+      LL_SKIP_REQUIRE_SYSROOT: 1
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
```

Comment:
I think this is what we want for making sure the python install is correctly setup.

> Using the setup-python action is the recommended way of using Python with GitHub Actions because it ensures consistent behavior across different runners and different versions of Python... For more information, see the [setup-python action](https://github.com/marketplace/actions/setup-python).

https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#specifying-a-python-version